src/modules/logicModule/abstracts/ERC20PaymentClientBase_v1.sol

```diff
@@ -236,9 +236,9 @@ abstract contract ERC20PaymentClientBase_v1 is
     }
 
     function _ensureValidPaymentOrder(PaymentOrder memory order) private view {
-        _ensureValidRecipient(order.recipient);
-        _ensureValidToken(order.paymentToken);
-        _ensureValidAmount(order.amount);
+        if (!(orchestrator().paymentProcessor().validPaymentOrder(order))) {
```

Comment:
The problem here is if for example the payment processor is the Simple, then orders are added, but then the payment processor of the Orchestrator is replaced, the time range will still not be validated.

In general, I think it is important to process all payments before changing a payment processor.

Reply:
We will address this by updating/rephrasing the disclaimer in the Natspec at the associated function.